bot/cogs/bot.py

```diff
@@ -60,6 +62,23 @@ def __init__(self, bot: AutoShardedBot):
 
         await ctx.invoke(self.info)
 
+    async def on_message(self, msg: Message):
+        if msg.content.count("\n") >= 3:
+            try:
+                tree = ast.parse(msg.content)
+                # Attempts to parse the message into an AST node.
+                # Invalid Python code will raise a SyntaxError.
+                if not all(isinstance(node, ast.Expr) for node in tree.body):
+                    # we dont want multiple lines of single words,
+                    # they would be syntactically valid Python but could also be
+                    # just some random multiline text someone is sending.
+                    formatted = f"```python\n{msg.content}\n```"
+                    if len(formatted) <= 2000:
+                        await msg.channel.send(formatted)
+                        await msg.delete()
+            except SyntaxError:
+                pass
+
```

Comment:
`await self.bot.process_commands()`
We don't want our beautiful commands to not work now do we?

Reply:
Actually put that at the start of on_message, not afterwards

Reply:
I was in belief that cog event listeners don't stop the bot from processing commands? Otherwise all on_message events would have to call that. Correct me if I'm wrong.

Reply:
on_message usually calls process_commands(), but he's overwriting on_message meaning he has to manually put process_commands there, I had to do the same with my bot (When doing everyone's favourite XP system).

Reply:
But bot.process_commands() will cause errors again wont it?

Reply:
@gdude2002 Hey, sorry to pull you into this, happen to know whether a cogs `on_message` function needs a `self.bot.process_commands(msg)` call?

Reply:
Er, it doesn't. We have several that don't have it.

Reply:
Just clarified with gdude, it's not required, ignore the process_commands function, un-needed.

Reply:
Ok my bad I was wrong

Reply:
Yes, i can read :wink:.